src/groupByKey.ts

```diff
@@ -1,7 +1,16 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
+type GroupsMap = {
+  [key: string]: [];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: GroupsMap[], key: keyof GroupsMap): object {
```

Comment:
You should read more about generic types in functions - https://mate-academy.github.io/fe-program/js/extra/typescript#use-type-as-a-function-parameter